official/benchmark/models/resnet_imagenet_main.py

```diff
@@ -37,6 +38,58 @@
 from official.vision.image_classification.resnet import resnet_model
 
 
+def selective_layers_to_cluster(model):
```

Comment:
I think these four functions (`selective_layers_to_cluster` to `get_selectively_clustered_model`) have unnecessary layers of indirection.

Something like this should suffice:

```
def cluster_last_three_conv2d_layers(model):
    last_three_conv2d_layers =  [
          layer for layer in model.layers
          if isinstance(layer, tf.keras.layers.Conv2D) and
          not isinstance(layer, tf.keras.layers.DepthwiseConv2D)
     ]
     
     cluster_centroids_init = 'linear'

     def cluster_fn(layer):
        if layer not in last_three_conv2d_layers:
           return layer

        if layer == last_three_conv2d_layers[0] or
           layer == last_three_conv2d_layers[1]:
               clustered = cluster.cluster_weights(layer, clusters=256, cluster_centroids_init=cluster_centroids_init)
        else:
             clustered = cluster.cluster_weights(layer, clusters=32, cluster_centroids_init=cluster_centroids_init)

        print("Clustered {} with {} clusters".format(layer.name, clustered.number_of_clusters)

     return tf.keras.models.clone_model(model, clone_function=cluster_fn)
     
```
1. There isn't a need to have different function arguments unless they are actually configurable by an end user (which isn't needed in this current version ... only the long-term integration). 
2. See https://google.github.io/styleguide/pyguide.html#310-strings.
3. I switched to `layer` instead of `layer.name`, which I don't think there are disadvantages, and it makes the code a bit cleaner.

Reply:
Thanks!

Reply:
Looks good. Resovling.